src/web/components/Core/Dialog.tsx

```diff
@@ -14,6 +14,7 @@ export type DialogProps = {
   onOpenChange?: (open: boolean) => void;
   fullScreen?: boolean;
   className?: string;
+  hideClose?: boolean;
```

Comment:
maybe `hideCloseContainer`? Or something that clearly conveys which thing we are hiding (rather than `hideCloseButton` since the `closeButton` is different). Maybe some other renaming required if you think of something reasonable.

Reply:
or `hideCloseContainerButton`?

Reply:
`hideCloseContainerButton` I like this more since it is actually button.

Reply:
actually just noticed that `closeButton` is a `string`, rather than a button. All uses of `closeButton` are just `'Cancel'`. Maybe we could refactor this so that your prop is indeed called `hideCloseButton` while `closeButton` is refactored to a boolean `showCancelButton` with a default text of `Cancel`?

Reply:
I think the operation should remain as it is currently. There are some uses where `closeButton` is empty but we still want the top right cross which is being hidden by `hideCloseContainerButton`. See `CreateAccountForm`, while it's quite weird implementation it's not using the default button. I wonder if the better solution would be to rename `closeButton` to `closeButtonText` and my new one `hideCloseButtons`, making it hide both the cross and the close button.

Reply:
This is what I was envisaging.

```
export type DialogProps = {
  triggerButton?: JSX.Element;
  children: ReactNode;
  title?: string;
  closeButtonText?: string;
  open?: boolean;
  onOpenChange?: (open: boolean) => void;
  fullScreen?: boolean;
  className?: string;
  hideCloseButtons?: boolean;
};
export function Dialog({
  triggerButton,
  children,
  title,
  closeButtonText,
  open,
  onOpenChange,
  fullScreen,
  className,
  hideCloseButtons = false,
}: DialogProps) {
  return (
    <RadixDialog.Root open={open} onOpenChange={onOpenChange}>
      {triggerButton && <RadixDialog.Trigger asChild>{triggerButton}</RadixDialog.Trigger>}
      <RadixDialog.Overlay className='dialog-overlay' />
      <RadixDialog.Content
        className='dialog-container'
        // To prevent dialog closed from user clicking outside of the dialog
        onInteractOutside={(fireEvent) => fireEvent.preventDefault()}
      >
        <div className={clsx({ fullScreen }, 'dialog-content', className)}>
          {title && <RadixDialog.Title className='dialog-title'>{title}</RadixDialog.Title>}
          {children}
          {closeButtonText && !hideCloseButtons && (
            <div className='dialog-close-button'>
              <RadixDialog.Close asChild>
                <button className='transparent-button' type='button'>
                  {closeButtonText}
                </button>
              </RadixDialog.Close>
            </div>
          )}
          {!hideCloseButtons && (
            <div className='dialog-close-container'>
              <RadixDialog.Close asChild>
                <button className='dialog-close-icon icon-button' aria-label='Close' type='button'>
                  <FontAwesomeIcon icon='xmark' />
                </button>
              </RadixDialog.Close>
            </div>
          )}
        </div>
      </RadixDialog.Content>
    </RadixDialog.Root>
  );
}
```
Sorry it's kinda a mess